pynamodb/models.py

```diff
@@ -975,3 +1022,28 @@ def _serialize_keys(cls, hash_key, range_key=None):
         if range_key is not None:
             range_key = cls._range_key_attribute().serialize(range_key)
         return hash_key, range_key
+
+
+class _ModelFuture:
+    """
+    A placeholder object for a model that does not exist yet
+
+    For example: when performing a TransactGet request, this is a stand-in for a model that will be returned
+    when the operation is complete
+    """
+    def __init__(self, model_cls):
+        self._model_cls = model_cls
+        self._model = None
+        self._resolved = False
+
+    def update_with_raw_data(self, data):
+        if data:
+            self._model = self._model_cls.from_raw_data(data=data)
+        self._resolved = True
+
+    def get(self):
+        if not self._resolved:
+            raise Exception('Cannot get until resolved')
```

Comment:
I think it's worth creating a new exception in pynamodb for this. You might want to model it off https://docs.python.org/3/library/asyncio-exceptions.html#asyncio.InvalidStateError, which is pretty similar. See https://docs.python.org/3/library/asyncio-task.html#asyncio.Task.result